src/Nethermind/Nethermind.Core/Crypto/Keccak.cs

```diff
@@ -5,22 +5,93 @@
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+
 using Nethermind.Core.Extensions;
 
 namespace Nethermind.Core.Crypto
 {
-    public unsafe struct ValueKeccak
+    [DebuggerStepThrough]
+    [DebuggerDisplay("{ToString()}")]
+    public readonly struct ValueKeccak : IEquatable<ValueKeccak>, IComparable<ValueKeccak>, IEquatable<Keccak>
     {
-        internal const int Size = 32;
-        public fixed byte Bytes[Size];
+        private readonly Vector256<byte> Bytes;
+
+        public const int MemorySize = 32;
 
-        public Span<byte> BytesAsSpan => MemoryMarshal.AsBytes(MemoryMarshal.CreateSpan(ref this, 1));
+        public Span<byte> BytesAsSpan => MemoryMarshal.AsBytes(MemoryMarshal.CreateSpan(ref Unsafe.AsRef(in Bytes), 1));
+
+        public ReadOnlySpan<byte> Span => MemoryMarshal.AsBytes(MemoryMarshal.CreateReadOnlySpan(ref Unsafe.AsRef(in Bytes), 1));
 
         /// <returns>
         ///     <string>0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470</string>
         /// </returns>
         public static readonly ValueKeccak OfAnEmptyString = InternalCompute(new byte[] { });
 
+        /// <returns>
+        ///     <string>0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347</string>
+        /// </returns>
+        public static readonly ValueKeccak OfAnEmptySequenceRlp = InternalCompute(new byte[] { 192 });
+
+        /// <summary>
+        ///     0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421
+        /// </summary>
+        public static readonly ValueKeccak EmptyTreeHash = InternalCompute(new byte[] { 128 });
+
+        /// <returns>
+        ///     <string>0x0000000000000000000000000000000000000000000000000000000000000000</string>
+        /// </returns>
+        public static ValueKeccak Zero { get; } = default;
+
+        /// <summary>
+        ///     <string>0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff</string>
+        /// </summary>
+        public static ValueKeccak MaxValue { get; } = new("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
+
+        public static implicit operator ValueKeccak(Keccak? keccak)
+        {
+            return new ValueKeccak(keccak?.Bytes);
+        }
+
+        public ValueKeccak(byte[]? bytes)
+        {
+            if (bytes is null || bytes.Length == 0)
+            {
+                Bytes = OfAnEmptyString.Bytes;
+                return;
+            }
+
+            Debug.Assert(bytes.Length == ValueKeccak.MemorySize);
+            Bytes = Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetArrayDataReference(bytes));
```

Comment:
can the byte array get collected/changed?

Reply:
What @LukaszRozmej raised. Why not copy?

Reply:
> can the byte array get collected/changed?

Passed in array if not used anywhere else can be GC'd or modified without effecting the Keccak

> Why not copy?

Is a by value copy of 32-byte type? (e.g. Vector256<byte>)

Reply:
Oh! I missed the assignment and focused on the cast 😓 . We're good here!